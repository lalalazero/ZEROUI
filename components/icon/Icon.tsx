import React from 'react'
import { classname } from '../_util/classes'
// import './importIcons'
import './Icon.scss'
// import '../icons/icons.js'

export const ICONS = [
    'like',
    'edit',
    'copy',
    'info',
    'error',
    'loading',
    'alipay',
    'wechat',
    'code-open',
    'code-close',
    'left',
    'right',
    'down',
    'setting',
    'emptysearch',
    'filled-up',
    'filled-down',
    'filled-left',
    'filled-right',
    'download',
    'next-double',
    'prev-double',
    'search',
    'zoom-in',
    'zoom-out',
    'close',
    'rotate-left',
    'rotate-right',
    'check',
    'check-filled',
    'happy',
    'joy',
] as const

const scriptElement = document.createElement('script')
scriptElement.src = 'http://at.alicdn.com/t/font_1353479_2q42sxdhtgu.js'
document.body.appendChild(scriptElement)

const prefix = 'zeroUI-icon'

export type ICON = typeof ICONS[number]
export interface IconProps extends React.SVGAttributes<SVGElement> {
    name: ICON
}
const Icon: React.FunctionComponent<IconProps> = ({
    className,
    name,
    ...restProps
}) => {
    return (
        <svg className={classname(className, prefix)} {...restProps}>
            <use xlinkHref={`#icon-${name}`}></use>
        </svg>
    )
}

export default Icon
