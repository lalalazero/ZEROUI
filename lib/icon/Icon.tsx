import React from 'react'
import './importIcons'
import './Icon.scss'
import classnames from '../helpers/classnames'

interface IconProps extends React.SVGAttributes<SVGElement> {
    name: string;
}
const Icon: React.FunctionComponent<IconProps> = ({ className, name, ...restProps }) => {
    return (
        <svg className={classnames('zeroUI-icon', className)} {...restProps}>
            <use xlinkHref={`#${name}`}></use>
        </svg>
    )
}

export default Icon;