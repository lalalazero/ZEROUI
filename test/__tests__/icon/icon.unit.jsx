import React from 'react'
import renderer from 'react-test-renderer';
import { mount } from 'enzyme'
import Icon from '../../../components/icon/Icon'
describe('icon 测试', () => {
    it('name = wechat', () => {
        const icon = renderer.create(<Icon name="wechat"></Icon>)
        let tree = icon.toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('onClick event', () => {
        let n = 1
        const fn = () => {
            n = 2
        }
        const component = mount(<Icon name="alipay" onClick={fn}></Icon>)
        component.find('svg').simulate('click')
        expect(n).toEqual(2)
    })
    it('onClick event, jest.fn() 测试', () => {
        let fn = jest.fn()
        let fn2 = jest.fn()
        const component = mount(<Icon name="alipay" onClick={fn}></Icon>)
        component.find('svg').simulate('click')
        expect(fn).toBeCalled()
    })
})