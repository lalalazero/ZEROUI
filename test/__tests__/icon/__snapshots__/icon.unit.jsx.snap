// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon 测试 name = wechat 1`] = `
<svg
  className="zeroUI-icon"
>
  <use
    xlinkHref="#wechat"
  />
</svg>
`;
